core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2865,8 +2865,14 @@ private void getExpressionConditionsAsList(Condition condition, List<ExpressionC
                 do {
                     List<String> tempUserList = new ArrayList<>();
                     answer = ldapContext.search(escapeDNForSearch(searchBase), searchFilter, searchControls);
+                    /*
+                    The size of the expression condition is used to verify the type of filter operation since the up
+                    coming steps needs to verify whether this is a multi attribute scenario or single attribute
+                    scenario.
+                     */
                     if (answer.hasMore()) {
-                        tempUserList = getUserListFromSearch(isGroupFiltering, returnedAttributes, answer);
+                        tempUserList = getUserListFromSearch(isGroupFiltering, returnedAttributes, answer,
+                                isSingleAttributeFilterOperation(expressionConditions.size()));
```

Comment:
There will be an NPE if the list is null.
Is this being handled

Reply:
In here always an empty list will be passed.